requests/sessions.py

```diff
@@ -157,12 +157,11 @@ def resolve_redirects(self, response, stream=False, timeout=None,
             if response.is_permanent_redirect and request.url != prepared_request.url:
                 self.redirect_cache[request.url] = prepared_request.url
 
-            self.rebuild_method(prepared_request, response)
+            method_changed = self.rebuild_method(prepared_request, response)
 
-            # https://github.com/kennethreitz/requests/issues/1084
-            if response.status_code not in (codes.temporary_redirect,
-                    codes.permanent_redirect):
-                # https://github.com/kennethreitz/requests/issues/3490
+            # https://github.com/kennethreitz/requests/issues/2590
```

Comment:
I think #3490 still makes sense to have, but #1094 is relating to code that is being removed here. We can keep it in if you think it's still providing value, but I feel it will end up as more of a red herring. Maybe making a note in #2590 with the link would be more useful if it's needed?